openwrt/autoflash

```diff
@@ -49,24 +54,28 @@ spawn gpioctl -p 3 1
 set timeout 30
 spawn tftp 192.168.1.1
 expect tftp>
-send "bin\n"
+send "bin\r"
 expect tftp>
-send "put factory.img\n"
+send "put factory.img\r"
 expect {
   Sent exp_continue
   "tftp>" {
-    send "quit\n\n"
+    send "quit\r"
   }
   "receive_packet: timeout" { exit 1 }
   "Try " { exit 1 }
 }
-close
 # Set back to default
 set timeout 10
-spawn sudo /usr/sbin/arp -d 192.168.1.1
-spawn sudo ifconfig ue1.503 create
-spawn sudo ifconfig ue1.503 inet 192.168.254.1 255.255.255.0
-spawn sudo dnsmasq -i ue1.503 --dhcp-range=192.168.254.100,192.168.254.100,255.255.255.0,30s --dhcp-option=3,192.168.254.1
+spawn arp -d 192.168.1.1
+spawn ifconfig flash0.503 create
+spawn ifconfig flash0.503 inet 192.168.254.1 255.255.255.0
+sleep 2
+
+# Our DHCP server to the AP
+# This will fail without the full path to dnsmasq, unsure why
```

Comment:
Im going to spin out a followup issue to see why this was an issue. But this would error with `file not found` and `PATH` seemed to be completely fine from what I could tell. Also running this locally on the pi `sudo -s ./autoflash` worked fine, so seems like it might be some side-effect due to the CI execution.

Reply:
weird… it’ll be an interesting find I'm sure

Reply:
Usually issues like this turn out to be environmental differences in my experience. (Either differences in the actual environment variables or differences in the execution environment (e.g. missing dependencies or files in different places than expected). They are often hard to debug and usually result in a "DOH!" forehead slap once identified.

As such, I think a quickly obtained workaround is probably good for now.